src/renderers/WebGLRenderer.js

```diff
@@ -65,7 +65,9 @@ function WebGLRenderer( parameters ) {
 		_antialias = parameters.antialias !== undefined ? parameters.antialias : false,
 		_premultipliedAlpha = parameters.premultipliedAlpha !== undefined ? parameters.premultipliedAlpha : true,
 		_preserveDrawingBuffer = parameters.preserveDrawingBuffer !== undefined ? parameters.preserveDrawingBuffer : false,
-		_powerPreference = parameters.powerPreference !== undefined ? parameters.powerPreference : 'default';
+		_powerPreference = parameters.powerPreference !== undefined ? parameters.powerPreference : 'default',
+		_webglversion = parameters.webgl2 !== undefined  ? 'webgl2' : 'webgl';
```

Comment:
Instead of adding anew parameter, can we try doing this for now?

```js
var canvas = document.createElement( 'canvas' );
var context = canvas.getContext( 'webgl2' );

var renderer = new THREE.WebGLRenderer( { context: context } );
```

Then, inside the renderer we can do:

```js
var isWebGL2 = _context instanceof WebGL2RenderingContext;
```

Reply:
Passing a context created outside does work already.  And you don't have to passing canvas even.  a canonical check for isWebGL2 at one place presumably at WebGLRenderer would be good.  (How do we proceed?  Would you prefer a fresh pull request or it is okay for me to push a new commit to this branch and you pull them?  Or this is close enough to be pulled, and we can fix issues over time?)

Reply:
> How do we proceed? Would you prefer a fresh pull request or it is okay for me to push a new commit to this branch and you pull them? Or this is close enough to be pulled, and we can fix issues over time?

Yeah, push changes to this branch.

I'm trying to see what's the minimum amount of changes needed in the current code to make this work.

Reply:
And I now remember why those isWebGL2 tests are not as simple as (_context instanceof WebGL2RenderingContext): it did not work on browsers that don't support WebGL2.  And also the renderer could potentially have a method called isWebGL2 or such, but the other classes such as WebGLExtensions.js, WebGLTextures.js, etc. only receives the actual context object and they still have to check it anyway.  (And actually, it'd be very helpful if you agree to keep webgl2 parameter to the WebGLRenderer.  It won't get in the way of others, right?)

I pushed a commit with the depth_frag error and just a cosmetic change to make the isWebGL2 check the same everywhere.